pepper-apis/src/main/java/org/broadinstitute/ddp/filter/SendGridEventVerificationFilter.java

```diff
@@ -43,6 +44,20 @@
 
     public SendGridEventVerificationFilter(String cfgParamSendGridEventsVerificationKey) {
         this.cfgParamSendGridEventsVerificationKey = cfgParamSendGridEventsVerificationKey;
+        registerSecurityProvider();
+    }
+
+    private void registerSecurityProvider() {
+        boolean alreadyRegistered = false;
+        for (var provider : Security.getProviders()) {
+            if (BouncyCastleProvider.PROVIDER_NAME.equals(provider.getName())) {
+                alreadyRegistered = true;
+                break;
+            }
+        }
+        if (!alreadyRegistered) {
+            Security.addProvider(new BouncyCastleProvider());
```

Comment:
We already have Bouncy Castle (BC) as a dependency, so no changes to POM files needed. I admit I don't know much about BC, so let me know if you have advice here!

Reply:
I looked some articles about BC usage as a provider in production - I haven't found any problems described.
The only concern: once BC is set it this filter then it will be used in DDP backend in all subsequent code. But as it is supports standard algorithms then it should not be a problem?

Reply:
Interesting. I had not considered the fact that BC will be used for the whole backend once this is set. I don't have any insight into whether this is a good thing or if there's other side-effects. Unfortunately, I don't know of a way to restrict this security provider to _just_ this sendgrid filter, so I think I will have to proceed with this as-is.